### PR TITLE
Add support for durations

### PR DESCRIPTION
**Pull request by webpapaya**
This is a work in progress PR which will add support for durations to date-fns. It will mostly migrate functions from https://www.npmjs.com/package/pomeranian-durations to date-fns in order to create one comprehensive library to work with dates, times and durations.

closes https://github.com/date-fns/date-fns/issues/277
closes https://github.com/date-fns/date-fns/issues/284

### CODE REVIEW

**Comment by leshakoss** [comment]
@webpapaya hello! Thanks a lot for your awesome contribution!

I've finally decided to check your PRs 😅 More detailed review is coming, but first, what is the difference between #348 and your other PR #364 ? I see that they are overlapping (many functions are in both). Can you make them a single PR?

Also, we decided to include durations as a feature of v2.0.0 which is coming very soon. Could you please rebase your branch to `v2` and reopen the PR to `date-fns:v2`? Thanks!

**Comment by azz0r** [comment]
Any update on this? would be great to get this in 👍

File: src/add_duration/test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+describe('addDuration', function () {
+  it('adds P1Y to date', function () {
+    var result = addDuration(new Date('2000-01-01T00:00:00'), 'P1Y')
+    assert.deepEqual(result, new Date('2001-01-01T00:00:00'))
+  })
+
+  it('adds P1M to date', function () {
+    var result = addDuration(new Date('2000-01-01T00:00:00'), 'P1M')
+    assert.deepEqual(result, new Date('2000-02-01T00:00:00'))
```
Comment (chamini2) [inline src/add_duration/test.js]: I'm curious of doing `addDuration(new Date('2001-01-30T00:00:00'), 'P1M')`. Would that result in `new Date('2000-02-29T00:00:00')` or in `new Date('2000-03-01T00:00:00')`?

Postgres goes for 02-29:

```psql
SELECT '2000-01-30'::date + 'P1M'::interval;
      ?column?
---------------------
 2000-02-29 00:00:00
```

**Comment by leshakoss** [comment]
Closing due to the fact that there were so much API changes since then that it's probably easier to rewrite this feature from scratch. I am sorry :(

**Pull request closed without merging**